off-chain/test/Examples/Addition/ContractSeath.purs

```diff
@@ -24,58 +27,105 @@ import Seath.Test.Examples.Addition.Actions
+  \(admin /\ u1 /\ u2 /\ u3) -> do
+    -- contract initialization by some admin
+    firstTransactionId /\ _ <- withKeyWallet admin initialContract
+
+    -- Seath round logic
+    vaildatorHash <- fixedValidatorHash
+    let leader = Leader u1
+    leaderPublicKeyHash <- getPublicKeyHash u1
     let
-      leader = Leader a
-      namedWallets = [ a /\ "Leader", b /\ "UserB", c /\ "UserC" ]
-      participants = map Participant [ b, c ]
+      participants = map Participant [ u2, u3 ]
       seathConfig = SeathConfig
         { leader: leaderPublicKeyHash
         , finalizedTxHandler: handleActionFromFinalizedTransaction
         , onchainHandler: handleActionFromBlockChain
         }
-    logBlockchainState namedWallets fixedValidatorHash
-    firstTransactionId /\ _ <- withKeyWallet a initialContract
-    logBlockchainState namedWallets fixedValidatorHash
+      logState = logBlockchainState leader participants vaildatorHash
+
+    logState -- log state after initialisation
+
     actions <- SeathSetup.genUserActions participants
-    logInfo' $ "test " <> show actions
+    logInfo' $ "User actions: " <> show actions
+
     let
       firstBuilderState = ChainBuilderState
         { finalizedTransactions: []
         , lastResult: Left firstTransactionId
         , pendingActions: actions
         }
-      buildChain =
-        actions2TransactionsChain
-          seathConfig
-          firstBuilderState
-    (finalizedTxsAndActions /\ _) <- withKeyWallet a buildChain
+      buildChain = actions2TransactionsChain seathConfig firstBuilderState
+
+    (finalizedTxsAndActions /\ _) <- withKeyWallet u1 buildChain
     let finalizedTxs /\ _ = unzip finalizedTxsAndActions
     -- logInfo' $ "BuildChainResult: " <> show finalizedTxs
-    _ <- SeathSetup.submitChain leader participants finalizedTxs $
-      logBlockchainState namedWallets fixedValidatorHash
-    logBlockchainState namedWallets fixedValidatorHash
+    txIds <- SeathSetup.submitChain leader participants finalizedTxs logState
+
+    case last txIds of
+      Nothing -> throwContractError
+        "No IDs vere received after chain submission. Something is wrong."
+      Just txId -> do
+        awaitTxConfirmed txId
+        logState
+        checkFinalState leader participants vaildatorHash
+
     logInfo' "end"
   where
 
-  distribution :: Array BigInt /\ Array BigInt /\ Array BigInt
+  distribution :: Array BigInt /\ Array BigInt /\ Array BigInt /\ Array BigInt
   distribution =
     [ BigInt.fromInt 1_000_000_000 ]
       /\ [ BigInt.fromInt 1_000_000_000 ]
       /\ [ BigInt.fromInt 1_000_000_000 ]
+      /\ [ BigInt.fromInt 1_000_000_000 ]
+
+checkFinalState :: Leader -> Array Participant -> ValidatorHash -> Contract Unit
+checkFinalState leader participants vaildatorHash = do
+  (BlockhainState bchState) <- getBlockhainState leader participants
+    vaildatorHash
+
+  checlLeaderUtxos bchState
+  checkScriptState bchState
+
+  where
+  checlLeaderUtxos bchState = do
```

Comment:
we also need a check that we aren't changing the leader utxos.

In the future when we add the fee for the leader, this would be broken, but as the fee mechanism is not clear right now, having this test is good for now.

Reply:
Maybe we can make something more general like `assertState` where we can pass predicates and check intermediate states also.

Reply:
yes, we definitely want something like that and also integrate the tests using mote.

Reply:
Made separate issue for this - #20